src/components/Progress/Progress.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+const Progress = ({
+  assistiveText,
+  children,
+  progress,
+  flavor,
+  isVertical,
+}) => {
+  const childrenCount = React.Children.count(children);
+  const progressPercent = progress / (childrenCount - 1) * 100;
+
+  return (
+    <div>
+      <div
+        className={cx('slds-progress', {
+          [`slds-progress_${flavor}`]: flavor,
+          'slds-progress_vertical': isVertical,
+        })}
+      >
+        <ol className="slds-progress__list">
+          {React.Children.map(children, child => React.cloneElement(child, { isVertical }))}
```

Comment:
@fspoettel What's the cleanest way to pass the `isVertical` prop to each child?

Reply:
This is the standard way to do it. You could go the extra mile to support `Fragments` (see `ButtonGroup` for an example for how to do it)

Reply:
Since there is other logic depending on the child count though, I'd say we better disallow supplying `Fragments`, what do you think?

Reply:
Would you enforce that via a custom PropType that checks on the children type and/or filter out "invalid" children?

Reply:
I'd just document it for now